File: 백준/동적계획법/9465.py
import sys

def solution(w):
    m = []

    m.append(list(map(int, sys.stdin.readline().split())))
    m.append(list(map(int, sys.stdin.readline().split())))

    m[0][1] += m[1][0]
    m[1][1] += m[0][0]

    for i in range(2, w):
        m[0][i] += max(m[1][i-1], m[1][i-2])
        m[1][i] += max(m[0][i-1], m[0][i-2])
    
    print(max(m[1][w-1], m[0][w-1]))

if __name__ == "__main__":
    T = int(input())
    for _ in range(T):
        solution(int(input()))

File: 백준/js/9465.js
// 백준 9465. 스티커

/* ========================== 백준 입력 ========================== */
// let fs = require("fs");
// let input = fs.readFileSync("/dev/stdin").toString();
/* ============================================================== */

function solution(n, arr) {
	arr[0].unshift(0);
	arr[1].unshift(0);

	for (var i = 2; i < n + 1; i++) {
		arr[0][i] += Math.max(arr[0][i - 2], arr[1][i - 2], arr[1][i - 1]);
		arr[1][i] += Math.max(arr[0][i - 2], arr[1][i - 2], arr[0][i - 1]);
	}

	console.log(Math.max(arr[0][n], arr[1][n]));

	return;
}

const caseNumber = parseInt(input.shift());

for (var i = 0; i < caseNumber; i++) {
	let n = parseInt(input.shift());

	if (n === 1) {
		const upElement = parseInt(input.shift());
		const downElement = parseInt(input.shift());
		const result = Math.max(upElement, downElement);

		console.log(result);
		continue;
	}

	const arr = [
		input
			.shift()
			.split(" ")
			.map((_) => parseInt(_)),
		input
			.shift()
			.split(" ")
			.map((_) => parseInt(_)),
	];

	solution(n, arr);
}
